lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/StateMachine/TaskStateTransitionException.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.Globalization;
+
+namespace Org.Apache.REEF.IMRU.OnREEF.Driver.StateMachine
+{
+    /// <summary>
+    /// Exception for TaskState Transition
+    /// </summary>
+    internal sealed class TaskStateTransitionException : Exception
```

Comment:
The purpose of the exception is not only for a new type but also for getting event and state as input so that to build exception message in a consistent way. 
I can inherit from that one if it adds any value.